src/pawelak-arek/Fsd.Pawelak.Cs/Fsd.Arek.Cs3/Domain/Warehouse.cs

```diff
@@ -53,11 +46,47 @@ public void AddProductToWareHouse(int count, Product product)
 
     public void SellProduct(int count, Product product)
     {
-        int uniqeCount = GetUniqecount(product.ProductId, _stock);
+        int uniqeCount = GetUniqecount(product.ProductId, WarehouseState);
 
         if (uniqeCount >= count)
             for (int i = 0; i < count; i++)
-                _stock.Remove(product);
+                WarehouseState.Remove(product);
+    }
+
+    public int CreateProductID(string name, decimal price)
+    {
+        int id = 0;
+        bool state;
+
+        try
+        {
+            state = WarehouseState.Exists(x => x.Name == name) && WarehouseState.Exists(x => x.Price == price) ? true : false;
+        }
+        catch
```

Comment:
to jest niezalecane. Polici dowolny wyjątek, tu go przechwycisz i stracisz o nim informacje.